pkg/apis/admissionregistration/types.go

```diff
@@ -123,8 +124,82 @@ type ValidatingAdmissionPolicy struct {
 	metav1.ObjectMeta
 	// Specification of the desired behavior of the ValidatingAdmissionPolicy.
 	Spec ValidatingAdmissionPolicySpec
+	// The status of the ValidatingAdmissionPolicy, including warnings that are useful to determine if the policy
+	// behaves in the expected way.
+	// Populated by the system.
+	// Read-only.
+	// +optional
+	Status ValidatingAdmissionPolicyStatus
+}
+
+// ValidatingAdmissionPolicyStatus represents the status of an admission validation policy.
+type ValidatingAdmissionPolicyStatus struct {
+	// The generation observed by the controller.
+	// +optional
+	ObservedGeneration int64
+
+	// TypeChecking contains type check warnings per expression.
+	// Presence of this field indicates the completion of the type checking.
+	// +optional
+	TypeChecking *TypeChecking
+
+	// Conditions contains the latest available observations of a policy's current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []ValidatingAdmissionPolicyCondition
+}
+
+// ValidatingAdmissionPolicyCondition describes the state of an admission validation policy at a certain point.
+type ValidatingAdmissionPolicyCondition struct {
+	// Type of deployment condition.
+	Type ValidatingAdmissionPolicyConditionType
+	// Status of the condition, one of True, False, Unknown.
+	Status api.ConditionStatus
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time
+	// The reason for the condition's last transition.
+	Reason string
+	// A human-readable message indicating details about the transition.
+	Message string
 }
 
+// ValidatingAdmissionPolicyConditionType is the condition type of admission validation policy.
+type ValidatingAdmissionPolicyConditionType string
+
+// TypeChecking contains results of type checking the expressions in the
+// ValidatingAdmissionPolicy
+type TypeChecking struct {
+	// ExpressionWarnings contains type checking warnings for each expression.
+	// +optional
+	// +listType=atomic
+	ExpressionWarnings []ExpressionWarning
+}
+
+// ExpressionWarning is a warning information that targets a specific expression.
+type ExpressionWarning struct {
+	// Expression is the CEL expression that this warning is associated with.
+	Expression string
+
+	// RefField indicates which field refers the expression.
+	RefField string
+
+	// Warning contains information about the expression in a human-readable form.
+	// Each line of the warning contains the type that the expression is checked
+	// against, followed by the type check error from the compiler.
+	Warning string
+}
+
+// ExpressionRefFieldType enumerates possible fields that can refer to an
+// expression
+type ExpressionRefFieldType string
+
+// ExpressionRefFieldValidation indicates that the expression is used as the
+// main expression of a validation.
+const ExpressionRefFieldValidation = "validation"
```

Comment:
```suggestion
const ExpressionRefFieldValidation = "validations"
```

To exactly match the field name in the resource?